### PR TITLE
Fix some issues with Pkg.pin, add some tests

### LINKED ISSUE #13520: libgit2 code typos
```
julia> Pkg.pin("Compat")
ERROR: UndefVarError: head_oid not defined
 in anonymous at pkg/entry.jl:299
 in with at ./libgit2/types.jl:481
 in pin at pkg/entry.jl:297
 in pin at pkg/entry.jl:307
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in pin at pkg.jl:46
```

### PR DESCRIPTION
**Pull request by jakebolewski**
closes #13520

### CODE REVIEW

File: base/libgit2/blob.jl [hunk 1]
```diff
@@ -5,7 +5,7 @@ function content(blob::GitBlob)
 end
 
 function isbinary(blob::GitBlob)
-    return ccall((:git_blob_is_binary, :libgit2), CInt, (Ptr{Void},), blob.ptr) == 1
+    return ccall((:git_blob_is_binary, :libgit2), Cint, (Ptr{Void},), blob.ptr) == 1
```
Comment (tkelman) [inline base/libgit2/blob.jl]: is this hit by any tests?

File: base/libgit2/merge.jl [hunk 2]
```diff
@@ -80,7 +80,7 @@ function merge!(repo::GitRepo, anns::Vector{GitAnnotated},
     anns_size = Csize_t(length(anns))
     @check ccall((:git_merge, :libgit2), Cint,
                   (Ptr{Void}, Ptr{Ptr{Void}}, Csize_t,
-                   Ptr{MergeOptionsStruct}, Ptr{CheckoutOptions}),
+                   Ptr{MergeOptions}, Ptr{CheckoutOptions}),
```
Comment (tkelman) [inline base/libgit2/merge.jl]: likewise here

File: base/libgit2/blob.jl (hunk 1 above)
Comment (jakebolewski) [inline base/libgit2/blob.jl]: they are not used at all

Comment (tkelman) [inline base/libgit2/blob.jl]: remove then?

Comment (wildart) [inline base/libgit2/blob.jl]: This one is unused.

File: base/libgit2/merge.jl (hunk 2 above)
Comment (jakebolewski) [inline base/libgit2/merge.jl]: @wildart this version of merge is not used by Pkg or PkgDev

Comment (wildart) [inline base/libgit2/merge.jl]: And we never use proper merge, so no tests

File: base/libgit2/blob.jl (hunk 1 above)
Comment (jakebolewski) [inline base/libgit2/blob.jl]: I'm just going to delete them.  I had a PR about this a while back but that touched some functionality that wasn't tested.

File: base/libgit2/merge.jl (hunk 2 above)
Comment (tkelman) [inline base/libgit2/merge.jl]: update should do fast forwards or pull with rebase, so should this function also be deleted and/or moved to a different repo?

**Pull request merged**